src/component/project/ProjectAccess/ProjectAccessAssign/ProjectAccessAssign.tsx

```diff
@@ -177,7 +173,8 @@ export const ProjectAccessAssign = ({
             ) || [];
         setSelectedOptions(selectedOption);
         setRole(roles.find(({ id }) => id === selected?.entity.roleId) || null);
-    }, [open, selected, options, roles]);
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [open, selected, JSON.stringify(options), JSON.stringify(roles)]);
```

Comment:
I think this will still create a problem if someone changes the roles in the system.